agents/demo/demo.ipynb

```diff
@@ -0,0 +1,970 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set the variables below to a project and bucket suitable for your use."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# GCP project to use\n",
+    "PROJECT=\"kubeflow-rl\"\n",
+    "\n",
+    "# Bucket to use\n",
+    "BUCKET=PROJECT+\"-kf\"\n",
+    "\n",
+    "# K8s cluster to use\n",
+    "CLUSTER=\"kubeflow-5fc52116\"\n",
+    "ZONE=\"us-east1-d\"\n",
+    "NAMESPACE=\"rl\"\n",
+    "\n",
+    "# Find the path to the workspace ml-app dir \n",
+    "import subprocess, os\n",
+    "def find_workspace_root(cwd=None):\n",
+    "    if cwd is None:\n",
+    "        cwd = subprocess.check_output([\"pwd\"]).strip()\n",
+    "    files = subprocess.check_output([\"ls\", cwd])\n",
+    "    if \"app.yaml\" in files:\n",
+    "        return cwd\n",
+    "    else:\n",
+    "        cwd = '/'.join(os.path.split(cwd)[:-1])\n",
+    "        return find_workspace_root(cwd)\n",
+    "APP_ROOT=find_workspace_root()\n",
+    "\n",
+    "# Needed for launching tensorboard\n",
+    "SECRET_NAME = \"gcp-credentials\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Attention:** You will need GCP credentials to access the cluster and GCP resources.\n",
+    "\n",
+    "If you're running this on your local machine you can authenticate for the GCP project you specified above in the usual way (i.e. `gcloud auth login` followed by `gcloud config set project <your-project-name>`; in this case skip the following step.\n",
+    "\n",
+    "If you're running this on JupyterLab you can do the following to provide the right credentials:\n",
+    "\n",
+    "- Create a service account with the appropriate roles and download the private key\n",
+    "- Use JupyterLab to upload the service account to your pod\n",
+    "- Set the path to your service account in the cell below and then execute it to activate the service account"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "KEY_FILE=\"/Users/cb/Downloads/kubeflow-rl-ec0f4f646339.json\"\n",
+    "!gcloud auth activate-service-account --key-file={KEY_FILE}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!gsutil mb -p {PROJECT} gs://{BUCKET}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!gcloud container clusters --project={PROJECT} --zone={ZONE} get-credentials {CLUSTER}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!kubectl create namespace {NAMESPACE}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Download and install ksonnet if needed"
```

Comment:
Can we run this notebook in a Docker container with the right packages installed? (That same container could be used with JupyterLab)